### ISSUE TITLE
 Suspense freezes the page and does not show the fallback component - Amplify and Next 13-14

### ISSUE THREAD

**Issue by vitoorfraga** [aws-amplify/amplify-hosting#3843]
### Before opening, please confirm:

- [x] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React, Next.js

### Amplify APIs

Not applicable

### Amplify Categories

hosting

### Environment information

No environment variable is required to reproduce this test.

### Describe the bug

The issue occurs every time I make a request via SSR and use the Suspense API. The page is frozen, and under no circumstances is the component inside the fallback rendered.

<details>

```
export default function ChargePoints({ searchParams }) {
  const searchString = searchParams.search || ''

  return (
    <>
      <PageHeader title="Gerenciar pontos de recargas " />

      <section className={styles.wrapper}>
        <Search value={searchString} path="/operator/chargepoints" />

        <Suspense key={searchString} fallback={<LoadingAlert />}>
          <AdminChargePointsList searchString={searchString} />
        </Suspense>
      </section>
    </>
  )
}
```

</details>

The issue does not occur when deploying the project on Vercel.

### Expected behavior

This is a video of the application running within the Vercel environment.

https://github.com/aws-amplify/amplify-js/assets/80723488/74f4e28f-cc9e-48b1-8d38-a7633e93e333

This is a video of the application running within the AWS Amplify environment.

https://github.com/aws-amplify/amplify-js/assets/80723488/781f95e2-8eb0-4936-bc3d-4e32bdaf0f6e

### Reproduction steps

1- Perform an SSR request.
2- Use the Suspense API around the component that makes the fetch.
3- Deploy the application within Amplify.

**Comment by leiarenee** [comment]
We are having the same issue. Suspense does not work with amplify under nextjs

**Comment by syed-yawar** [comment]
This is related to streaming which works with Suspense. When app deployed on Amplify, streaming does not work. It might buffered the request somewhere and then after that passes it to broswer. 
details are mentioned [here](https://github.com/vercel/next.js/issues/54667#issuecomment-1745841559)
I'm still looking into the solution.

**Comment by aavaldez64** [comment]
I also have this problem, but even with Vercel and localhost, <Suspense> and loading.tsx file doesn't work. I'm using next v14.1.0 but also tried with next v14.0.2 (following the official tutorial https://nextjs.org/learn/dashboard-app/streaming) and the same thing happens. Suspense freezes the whole page until the component finishes loading.

**Comment by leiarenee** [comment]
> This is related to streaming which works with Suspense. When app deployed on Amplify, streaming does not work. It might buffered the request somewhere and then after that passes it to broswer. details are mentioned [here](https://github.com/vercel/next.js/issues/54667#issuecomment-1745841559) I'm still looking into the solution.

Looking forward to your solution.

**Comment by oczkowski** [comment]
I've done some digging around and it seems like the amplify host doesn't allow for streaming data in anyway. I've made POCs to prove this with API routes and SSR components.

Am I right to assume, Amplify runs the SSR components and serves them using lambda?

If so, by default lambdas do not allow for data streaming.
https://docs.aws.amazon.com/lambda/latest/dg/configuration-response-streaming.html

Amplify Gen 2 uses cloudformation for each app, I might look into that if my assumption is correct.

**Comment by oczkowski** [comment]
I've done some digging around on how Amplify Gen 2 works. SSR components run off lambda using the same node version defined in amplify.yml file.

However, amplify hides the lambda's as part of the Amplify service, meaning streaming can't be enabled manually. The only thing you get to see is the runtime logs in CloudWatch under `/aws/amplify` log groups.

I'm meeting our AWS account manager today, I'll ask if anything can be done to fix this.

**Comment by syed-yawar** [comment]
Certainly, it all makes sense, but at the same time, it's quite frustrating to invest so much time in this.

Looking forward to your discussion, and hopefully, we can identify a workaround.

**Comment by oczkowski** [comment]
Haven't got much from the call, we're waiting for an amplify specialist to look into it.

**Comment by shawngustaw** [comment]
Anyone have any updates on this? Noticing `loading.tsx` works but manually creating Suspense boundaries doesn't work.

**Comment by oczkowski** [comment]
I know this isn't related to amplify hosting but for those looking for a solution, you can try using SST. It has an experimental but not production-ready solution for streaming https://docs.sst.dev/constructs/NextjsSite#experimentalstreaming

You can read more about the state of it here. https://open-next.js.org/inner_workings/streaming.

SST is similar to Amplify in that it uses CDK to build the cloud formation and all resources. The nice thing about it is that you can actually see all the resources. Both the lambda's and edge functions.

**Comment by syed-yawar** [comment]
Yes, you're right @oczkowski. I already have a plan to test sst serverless when I get the time. Additionally, we can also consider using [Flight Control for deployment](https://app.flightcontrol.dev/). While it's not serverless, it sets up everything on AWS via CloudFormation. The streaming and options are working correctly with this solution too.

**Comment by Jay2113** [comment]
Hi everyone 👋 , Amplify Hosting currently does not support Next.js streaming. To learn more about the Next.js feature support, refer to our documentation here: https://docs.aws.amazon.com/amplify/latest/userguide/ssr-Amplifysupport.html#supported-unsupported-features. 

We are tracking this as a feature request and will keep you posted on this thread with any updates.

**Comment by costleya** [comment]
This is a solved problem. https://github.com/awslabs/aws-lambda-web-adapter/tree/main/examples/nextjs-response-streaming

Is the reason cost on the AWS hosting side?

**Comment by leiarenee** [comment]
Amplify supports connection to lambda. Might there be a way for proxying partially to evaluate server-side part?

**Comment by romatallinn** [comment]
Yikes. Still an issue.

**Comment by mauerbac** [comment]
This is a feature request for Next.js streaming

**Comment by lucaloiacono** [comment]
Anyone found a solution for this issue while the streaming feature is not available on Amplify?
Many companies sticks on the AWS ecosystem and switching to Vercel (even partially) is not an option.

**Comment by vitoorfraga** [comment]
Hello guys, I found an alternative solution that, although it doesn't use the Amplify service, worked for me and has worked for others.

https://sst.dev/docs/start/aws/nextjs

**Comment by amkac** [comment]
Will this feature be supported soon?

**Comment by raysubham** [comment]
Loading UI is definitely important for a modern app. Amplify team, can we please prioritize this ?

**Comment by DavisuaCoder** [comment]
I have deploy my nextjs 14.1.4 on amplify and the problem persist, any alternative?

**Comment by swaminator** [comment]
Amplify team here. This is on our roadmap for 2025.

**Comment by danverbs** [comment]
> Amplify team here. This is on our roadmap for 2025.

The sooner the better please 🙏🏻

**Comment by leiarenee** [comment]
It's really unbeleivable that AWS does not respond to such an important
issue for a year or so. 😃 tragicomic as well.

**Comment by pkyeck** [comment]
I see that this is "in progress" for like 2 months now – is there any way to see what the progress is and when to expect something? this feature is really important b/c otherwise the performance of pages with a lot of API calls on the server are really not usable 🙁

and can we help to make this happen faster in any way?

**Comment by mauerbac** [comment]
Hi all -- (@leiarenee , @pkyeck ) - this project is going to be paused for the meantime. We do plan to support Streaming, but can't provide a current ETA. Just want to be transparent about this so you can plan accordingly. 

I'm really sorry for the delay in communication and frustration by the news

**Comment by shawngustaw** [comment]
Why doesn't Amazon just contribute to https://github.com/opennextjs/opennextjs-aws like Cloudflare has done for their adapter?

**Comment by zain-23** [comment]
Did anyone find an alternative solution?

**Comment by danverbs** [comment]
> Did anyone find an alternative solution?

Using 'loading.jsx/tsx' still works, which is streaming just not using the <Suspense> boundary, so I've just constructed loading pages that match the layout of the real pages to simulate the {fallback} element of Suspense

**Comment by costleya** [comment]
> Hi all -- (@leiarenee , @pkyeck ) - this project is going to be paused for the meantime. We do plan to support Streaming, but can't provide a current ETA. Just want to be transparent about this so you can plan accordingly. 
> 
> I'm really sorry for the delay in communication and frustration by the news 

While that is incredibly unfortunate to hear, it's good to know AWS is at least thinking about this.

**Comment by realtebo** [comment]
We aboslutely need this feature. It's one of killer feature

The only workaround is move out of your shitty clone and return to vercel hosting !?!?!??!

**Comment by realtebo** [comment]
@vitoorfraga : i suggest to update title to include up to next 16.

 Here: https://github.com/orgs/aws-amplify/projects/130?pane=issue&itemId=69030136&issue=aws-amplify%7Camplify-hosting%7C3843 

This request is stille up for consideration and not even planned

**Comment by realtebo** [comment]
Please increase importance of this feature request.

**Comment by sudarshan-js** [comment]
It's surprising that this has not been worked on yet. We are facing the same issue in our org and will be considering moving to Vercel.

**Comment by hiddehe** [comment]
We're also facing this issue, considering SST or vercel rn. Especially with LLM responses streaming is a must.

**Comment by FlavioInacio-jf** [comment]
This issue has been open for two years, and we have nothing yet.

**Comment by speedhawk21** [comment]
Can someone please bump to the current/next sprint.  This has been open for way too long and we all would like Amplify Hosting to maintain feature parity with Vercel on a pretty common use case.

**Comment by crafael23** [comment]
This is just sad now...

**Comment by adrianjoshua-strutt** [comment]
Hey everyone. Quick update relevant to this thread. The Amplify Gen2 team just opened a developer-preview RFC for self-managed hosting that addresses what's being asked here. It deploys a CloudFront + Lambda stack in your account with streaming enabled. No Amplify Hosting console involved.

What this looks like in practice:

**1. Install**
```
npm install @aws-amplify/hosting@iac-hosting @aws-amplify/backend-cli@iac-hosting @aws-amplify/backend@iac-hosting
```

**2. Create `amplify/hosting.ts`**
```typescript
import { defineHosting } from '@aws-amplify/hosting';

defineHosting({
  framework: 'nextjs',
});
```

**3. Deploy**
```
npx ampx deploy --identifier my-app
```

RFC with full details and feedback thread: https://github.com/aws-amplify/amplify-backend/issues/3211

This is a developer preview. APIs may shift. This is a self-managed option separate from the existing Amplify Hosting console product. You own the stack, deploy with your own CI/CD or locally. Give it a shot and let us know how streaming behaves for your use case.

cc @vitoorfraga @leiarenee @syed-yawar @aavaldez64 @oczkowski @Jay2113 @shawngustaw @NickBlow @oliveiracdz @karpcs @mcfry @6mdx @SakataRyuichi @dragos99 @costleya @MrLightful @MasahiroKanatani @skorfmann @phyeony @giofilo @DoctorOak @caleb-sideras @CesarAcosta13
